src/main/java/com/fakng/fakngagrgtr/persistent/vacancy/VacancyRepository.java

```diff
@@ -17,18 +17,19 @@ public interface VacancyRepository extends CrudRepository<Vacancy, Long> {
     @Query(value = "select v from Vacancy v " +
             "join fetch v.company c " +
             "join fetch v.locations l " +
-            "where c.id in :companyIds and l.id in :locationIds",
+            "where c.id in :companyIds and l.id in :locationIds and lower(v.title) like lower(CONCAT(:title,'%'))",
```

Comment:
Make this search in middle of titles: `lower(CONCAT('%', :title, '%'))`

Reply:
I think you are right... I did it initially because postgres works slower (if index added) with pattern search with this %pattern% than with this pattern%

But functionally it's worse, of course

Besides, this thing made me think that this is a poor design overall. Kinda pattern search in DB. Maybe when we'll be introducing keywords indexing, it's worth indexing all vacancies and creating a search index for titles also... Using ElasticSearch or something

But I'll change this thing for sure, you are 100% correct